.github/workflows/codeql.yml

```diff
@@ -0,0 +1,48 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches:
+    - "main"
+  pull_request:
+    branches:
+    - "main"
+  schedule:
+  - cron: '16 11 * * 5'
+
+permissions:
```

Comment:
nit: im tryhing to make a ~standard of putting this first after the name - similar in jobs secrets/permissions first so they are more obvious